### PR TITLE
Lookup contestable issues by file number

### LINKED ISSUE #15451: Lookup contestable issues by file number
## Description
Currently when retrieving contestable issues you can only retrieve them by SSN. This issue would allow contestable issues to be able to be retrieved via file number. 

## Acceptance criteria
- [x] Update `BaseContestableIssuesController` to allow for file number header and retrieve veteran via file number
- [x] Update specs to include loading contestable issues via file number

### PR DESCRIPTION
**Pull request by templeman15**
Resolves #15451 

### Description
- [x] Update BaseContestableIssuesController to allow for file number header and retrieve veteran via file number
- [x] Update specs to include loading contestable issues via file number

### CODE REVIEW

File: app/controllers/api/v3/decision_reviews/base_contestable_issues_controller.rb [hunk 1]
```diff
@@ -41,10 +41,14 @@ def validate_params
...
             render_invalid_veteran_ssn
             return false
           end
+          if veteran_identifier.nil?
+            render_missing_headers
+            return false
+          end
           unless veteran
             render_veteran_not_found
             return false
```
Comment (edmangimelli) [inline app/controllers/api/v3/decision_reviews/base_contestable_issues_controller.rb:54]: what would you think about adding something like:
```
          if veteran_ssn && veteran_ssn != veteran.ssn
            render_veteran_not_found
            return false
          end
          if veteran_file_number && veteran_file_number != veteran.file_number
            render_veteran_not_found
            return false
          end
```
after this `unless veteran` block
?

Comment (templeman15) [reply app/controllers/api/v3/decision_reviews/base_contestable_issues_controller.rb:54]: I can do that. Update pushed!

File: app/controllers/api/v3/decision_reviews/base_contestable_issues_controller.rb [hunk 2]
```diff
@@ -41,17 +41,32 @@ def validate_params
...
+          if veteran_ssn.present? && !veteran_ssn_is_formatted_correctly?
             render_invalid_veteran_ssn
             return false
           end
-          unless veteran
+          if veteran_identifier.nil?
+            render_missing_headers
+            return false
+          end
+          unless veteran && veteran_matches
```
Comment (tomas-nava) [inline app/controllers/api/v3/decision_reviews/base_contestable_issues_controller.rb:52]:
```suggestion
          unless veteran.present? && veteran_matches
```
...to be consistent with the use of `present?`

File: app/controllers/api/v3/decision_reviews/base_contestable_issues_controller.rb [hunk 3]
```diff
@@ -41,17 +41,32 @@ def validate_params
...
             render_veteran_not_found
             return false
           end
           true
         end
 
+        def veteran_matches
+          if veteran_ssn && veteran_ssn != veteran.ssn
```
Comment (tomas-nava) [inline app/controllers/api/v3/decision_reviews/base_contestable_issues_controller.rb:60]:
```suggestion
          if veteran_ssn.present? && veteran_ssn != veteran.ssn
```
see previous note

File: app/controllers/api/v3/decision_reviews/base_contestable_issues_controller.rb [hunk 4]
```diff
@@ -41,17 +41,32 @@ def validate_params
...
+            return false
+          end
+          if veteran_file_number && veteran_file_number != veteran.file_number
```
Comment (tomas-nava) [inline app/controllers/api/v3/decision_reviews/base_contestable_issues_controller.rb:63]:
```suggestion
          if veteran_file_number.present? && veteran_file_number != veteran.file_number
```
see previous note

**Review by tomas-nava — APPROVED** [review]
approve!

with non-blocking suggestions

File: spec/requests/api/v3/decision_reviews/higher_level_reviews/contestable_issues_controller_spec.rb [hunk 5]
```diff
@@ -1,5 +1,9 @@
 # frozen_string_literal: true
 
+require_relative(
+  "../../../../../support/shared_examples/requests/api/v3/decision_reviews/shared_examples_contestable_issues.rb"
+)
+
```
[the remark this answers is not in the document]
Comment (templeman15) [reply spec/requests/api/v3/decision_reviews/higher_level_reviews/contestable_issues_controller_spec.rb]: @tomas-nava made this change and all of the other requested changes below, thanks!

**Pull request merged**